test/util/ChartUtils/getDomainOfErrorBars.spec.tsx

```diff
@@ -0,0 +1,288 @@
+      [undefined, 'radiusAxis', undefined],
+      [undefined, 'xAxis', 'x'],
+      [undefined, 'xAxis', undefined],
+      [undefined, 'yAxis', 'y'],
+      [undefined, 'yAxis', undefined],
+      [undefined, 'zAxis', undefined],
+      [undefined, undefined, 'x'],
+      [undefined, undefined, 'y'],
+      [undefined, undefined, undefined],
+    ];
+    const returnsNullCases: Case[] = [
+      ['centric', 'angleAxis', 'x'],
+      ['centric', 'angleAxis', 'y'],
+      ['centric', 'radiusAxis', 'x'],
+      ['centric', 'radiusAxis', 'y'],
+      ['centric', 'xAxis', 'y'],
+      ['centric', 'yAxis', 'x'],
+      ['centric', 'zAxis', 'x'],
+      ['centric', 'zAxis', 'y'],
+      ['horizontal', 'angleAxis', 'x'],
+      ['horizontal', 'angleAxis', 'y'],
+      ['horizontal', 'angleAxis', undefined],
+      ['horizontal', 'radiusAxis', 'x'],
+      ['horizontal', 'radiusAxis', 'y'],
+      ['horizontal', 'radiusAxis', undefined],
+      ['horizontal', 'xAxis', 'x'],
+      ['horizontal', 'xAxis', 'y'],
+      ['horizontal', 'xAxis', undefined],
+      ['horizontal', 'zAxis', 'x'],
+      ['horizontal', 'zAxis', 'y'],
+      ['horizontal', 'zAxis', undefined],
+      ['radial', 'angleAxis', 'x'],
+      ['radial', 'angleAxis', 'y'],
+      ['radial', 'radiusAxis', 'x'],
+      ['radial', 'radiusAxis', 'y'],
+      ['radial', 'xAxis', 'y'],
+      ['radial', 'yAxis', 'x'],
+      ['radial', 'zAxis', 'x'],
+      ['radial', 'zAxis', 'y'],
+      ['vertical', 'angleAxis', 'x'],
+      ['vertical', 'angleAxis', 'y'],
+      ['vertical', 'angleAxis', undefined],
+      ['vertical', 'radiusAxis', 'x'],
+      ['vertical', 'radiusAxis', 'y'],
+      ['vertical', 'radiusAxis', undefined],
+      ['vertical', 'yAxis', 'x'],
+      ['vertical', 'yAxis', 'y'],
+      ['vertical', 'yAxis', undefined],
+      ['vertical', 'zAxis', 'x'],
+      ['vertical', 'zAxis', 'y'],
+      ['vertical', 'zAxis', undefined],
+      [undefined, 'angleAxis', 'x'],
+      [undefined, 'angleAxis', 'y'],
+      [undefined, 'radiusAxis', 'x'],
+      [undefined, 'radiusAxis', 'y'],
+      [undefined, 'xAxis', 'y'],
+      [undefined, 'yAxis', 'x'],
+      [undefined, 'zAxis', 'x'],
+      [undefined, 'zAxis', 'y'],
+    ];
+
+    it('should cover all cases without duplicates', () => {
+      const set = new Set<string>();
+      function verify(c: Case) {
+        const s = JSON.stringify(c);
+        if (set.has(s)) {
+          throw new Error(`Test case duplicate: ${s}`);
+        }
+        set.add(s);
+      }
+      relevantCases.forEach(verify);
+      returnsNullCases.forEach(verify);
+      const countOfLayoutOptions = 5; // 4 layouts + undefined
+      const countOfAxisOptions = 6; // 5 axes + undefined
+      const countOfDirectionOptions = 3; // 2 directions + undefined
+      const allCombinationsCount = countOfLayoutOptions * countOfAxisOptions * countOfDirectionOptions;
+      expect(set.size).toBe(allCombinationsCount);
+    });
```

Comment:
This is just wonderful. I was about to start evaluating this myself manually. 😮‍💨

Reply:
I am contemplating adding a generator and creating these cases on the fly.